### PR TITLE
Product-listing-page

### PR DESCRIPTION
**Pull request by rahul31agrawal**
Hi Folks,
I have made a product listing page for my ecomm app.
* I have not implemented any functionality, I will add it in next commit.
Please give your PR reviews.
Thanks
productListingPage

### CODE REVIEW

File: src/pages/productListing/ProductListing.jsx [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+              <span className="icon-button__badge">2</span>
+            </button>
+          </div>
+        </div>
+      </nav>
+
+      <div className="content product-grid">
+        <div className="filters">
+          <div className="filter-heading">
+            <h3 className="filter-head main">Filters</h3>
```
Comment (sashaboi) [inline src/pages/productListing/ProductListing.jsx:72]: I suggest the filters on hover should have cursor pointer to give feel like button

File: src/pages/components/card/card.jsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+                <h6>{categoryName}</h6>
+              </div>
+              <div className="secondary-container1">
+                <p className="disc-price1">{discountedPrice}</p>
+                <p className="actual-price1">{price}</p>
+                <p className="offer-info1">{discount}% Off</p>
+              </div>
+              <div className="card-footer1">
+                <button className="btn-link1 call-to-action1 vertical-btn1">
+                  Add to cart
```
Comment (sashaboi) [inline src/pages/components/card/card.jsx:27]: hover cursor pointer here

File: src/pages/productListing/ProductListing.jsx [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+                name="price"
+                id="price"
+                min="0"
+                max="500000"
+                value="50"
+              />
+            </div>
+            <div className="filter-category --category">
+              <h3 className="filter-head">Category</h3>
+              <input type="checkbox" id="men" name="men" value="men" />
```
Comment (sashaboi) [inline src/pages/productListing/ProductListing.jsx:91]: the input and label should have some gap , they are stuck together

File: src/pages/components/card/card.jsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+
+export default function Card ({name,categoryName,price,discountedPrice,discount,stockDetail,cardImage}){
+return(
+    <div className="container1">
+              <div>
+                <img className="ecom-img1" src={cardImage} alt="ecomimage" />
+                <span className="card-badge1"> {stockDetail} </span>
+                <span className="heart-badge1">
+                  <i className="fas fa-heart"></i>
```
Comment (sashaboi) [inline src/pages/components/card/card.jsx:13]: this seems to be the wishlist button, so cursor pointer and on hover color change is imp

File: src/pages/components/card/card.jsx [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+//import cardImgage from "../../assets/images/MensFashion.jpeg";
+
+
+export default function Card ({name,categoryName,price,discountedPrice,discount,stockDetail,cardImage}){
+return(
+    <div className="container1">
+              <div>
+                <img className="ecom-img1" src={cardImage} alt="ecomimage" />
+                <span className="card-badge1"> {stockDetail} </span>
```
Comment (sashaboi) [inline src/pages/components/card/card.jsx:11]: the top left border radius is being messed up by this badge maybe add the border radius to badge too?

**Review by sashaboi — APPROVED** [review]
Looks good , the UI looks clean. Looking forward to future updates. 

Made a few non necessary suggestions you may wanna incorporate for better UX

File: src/App.js [hunk 6]
```diff
@@ -1,10 +1,16 @@
 import "./App.css";
+import { LandingPage } from "./pages";
+
+
+
 
 
 function App() {
   return (
     <div className="App">
```
Comment (vikrantkode) [inline src/App.js:10]: try to be consistent with the spaces or formatting. It increases readability and code also looks good

File: src/backend/db/products.js [hunk 7]
```diff
@@ -1,3 +1,10 @@
+import tshirtImage1 from "../../pages/assets/images/tshirt-img/Tshirt image1.jpg";
+import tshirtImage2 from "../../pages/assets/images/tshirt-img/Tshirt image2.jpg";
+import tshirtImage3 from "../../pages/assets/images/tshirt-img/Tshirt image3.jpg";
+import tshirtImage4 from "../../pages/assets/images/tshirt-img/Tshirt image4.jpg";
+import tshirtImage5 from "../../pages/assets/images/tshirt-img/Tshirt image5.jpg";
+import tshirtImage6 from "../../pages/assets/images/tshirt-img/Tshirt image6.jpg";
+
```
Comment (vikrantkode) [inline src/backend/db/products.js:7]: import all the images in one file and import from there

File: src/pages/components/card/card.jsx (hunk 5 above)
Comment (vikrantkode) [inline src/pages/components/card/card.jsx:11]: use camelcasing in alt

File: src/pages/productListing/ProductListing.jsx [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+    
+
+    axios.get('/api/products')
+    .then((response)=>{
+      setItem(response.data.products);
+
+    },
+    (error)=>{
+      console.log(error);
+
```
Comment (vikrantkode) [inline src/pages/productListing/ProductListing.jsx:21]: try to use try..catch format for good practise - (Personal Suggestion you can use this later after finishing e-comm)

File: src/pages/productListing/ProductListing.jsx [hunk 9]
```diff
@@ -0,0 +1,166 @@
...
+                id="lowtohigh"
+                name="sortby"
+                value="Low to High"
+              />
+              <label htmlFor="lowtohigh">Low To High</label>
+              <br />
+              <input
+                type="radio"
+                id="hightolow"
+                name="sortby"
```
Comment (vikrantkode) [inline src/pages/productListing/ProductListing.jsx:143]: Camelcasing is missing here. Its suggested to use everywhere in code for good practises

File: src/pages/productListing/product-Listing.css [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+    display: grid;
+    grid-template-columns: 1fr 1fr;
+    row-gap: 2.5rem;
+    column-gap: 1.5rem;
+    align-content: center;
+    justify-content: center;
+    align-items: center;
+    justify-items: center;
+} */
+
```
Comment (vikrantkode) [inline src/pages/productListing/product-Listing.css:73]: Try removing commented code while pushing

File: src/pages/productListing/product-Listing.css [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+    border: none;
+    text-align: end;
+    padding: 0.5rem;
+}
+
+.products-grid {
+    /* display: grid;
+    margin-top: 2rem;
+    grid-template-columns: 1fr 1fr 1fr;
+    gap: 1rem; */
```
Comment (vikrantkode) [inline src/pages/productListing/product-Listing.css:85]: remove commented code

**Review by vikrantkode — APPROVED** [review]
- Good folder structure and code is easily readable.
- some changes suggested, you can have look on them

**Pull request merged**